src/background/services/paymentSession.ts

```diff
@@ -300,4 +307,61 @@ export class PaymentSession {
     this.amount = amount.toString()
     this.intervalInMs = Number((amount * BigInt(HOUR_MS)) / BigInt(this.rate))
   }
+
+  private async getOverpayingSession(): Promise<{
+    session: OverpayingSession | null
+    sessionIndex: number
+  }> {
+    const { overpayingSessions } = await this.storage.get([
+      'overpayingSessions'
+    ])
+
+    const hashUrl = await getHashUrl(this.url || '')
+
+    const sessionIndex = overpayingSessions.findIndex(
+      (session) =>
+        session.tabId === this.tabId &&
+        session.hashUrl === hashUrl &&
+        session.walletAddress === this.receiver
+    )
```

Comment:
Can we make has this entirely as the key in storage? Can make key something like `{tabId}:{hashUrl}:{hashReceiver}`